encoder.go

```diff
@@ -32,8 +32,20 @@ func NewGenericStreamEncoder(w io.Writer, marshalFn Marshaler, array bool) *Stre
 	}
 }
 
+func (sw *StreamEncoder) SetFirst(first bool) *StreamEncoder {
+	sw.first = first
+	return sw
+}
+
+func (sw *StreamEncoder) SetArray(array bool) *StreamEncoder {
```

Comment:
This can be in the constructor rather than here, right?

Reply:
Hmm, yeah

Reply:
Actually I don't think so, I am doing this so I could override the default encoder settings here.
https://github.com/multiprocessio/datastation/blob/9de174a0a88d5623bfedf74601cd13a67c1e6f90/runner/file.go#L34